jenkins/Hono-Website-Pipeline-Declarative.groovy

```diff
@@ -69,150 +69,137 @@ pipeline {
-        sh '''
-            git clone --depth 1 --branch 4.2.5 https://github.com/McShelby/hugo-theme-relearn.git ${WORKSPACE}/hono-documentation-assembly/themes/hugo-theme-relearn
-        '''
-        echo "cloning Hugo Universal theme..."
-        sh '''
-            git clone --depth 1 --branch 1.3.2 https://github.com/devcows/hugo-universal-theme.git ${WORKSPACE}/hono/site/homepage/themes/hugo-universal-theme
-            echo "Remove images from theme" # We do not need the pictures. Removing them, so they don't get deployed
-            rm ${WORKSPACE}/hono/site/homepage/themes/hugo-universal-theme/static/img/*
-        '''
+        sshagent(["git.eclipse.org-bot-ssh"]) {
+          sh '''
+            #!/bin/bash
+            echo "cloning Hono web site repository..."
+            git clone ssh://genie.hono@git.eclipse.org:29418/www.eclipse.org/hono.git "${WEBSITE_TARGET_DIR}"
+            echo "scrubbing web site target directory..."
+            rm -rf "${WEBSITE_TARGET_DIR}"/*
+          '''
+        }
       }
     }
 
-    stage('Cloning Hono web site repository') {
+    stage("Build home page") {
       steps {
-        sshagent(["git.eclipse.org-bot-ssh"]) {
+        container("hugo") {
           sh '''
-            echo "cloning Hono web site repository..."
-            git clone ssh://genie.hono@git.eclipse.org:29418/www.eclipse.org/hono.git ${WORKSPACE}/hono-web-site
-            echo "scrubbing web site target directory..."
-            rm -rf ${WORKSPACE}/hono-web-site/*
+            #!/bin/bash
+            cd "${HONO_HOMEPAGE_DIR}"
+            echo "using $(hugo version)"
+            echo "removing obsolete images that come with theme used for home page ..."
+            # we do not need the pictures that come with the theme
+            # removing them, so they don't get deployed.
+            rm themes/hugo-universal-theme/static/img/*
+            echo "building home page..."
+            hugo -v -d "${WEBSITE_TARGET_DIR}"
           '''
         }
       }
     }
 
-    stage('Preparing web site source') {
+    stage("Build documentation") {
       steps {
-        echo "building documentation for versions"
-        sh '''#!/bin/bash
+        sh '''
+          #!/bin/bash
+          echo "determining supported versions of documentation"
           function prepare_stable {
-              VERSION="$1.$2"
-cat <<EOS >> $WORKSPACE/hono-documentation-assembly/config_version.toml
-  [Languages.stable]
-    weight = -20000
-    languageName = "stable ($VERSION)"
-    contentDir = "content_dirs/stable"
-    [Languages.stable.params]
-      honoVersion = "stable"
-EOS
-              git checkout $3
-              cp -r $WORKSPACE/hono/site/documentation/content $WORKSPACE/hono-documentation-assembly/content_dirs/stable
+            VERSION="$1.$2"
+            {
+              echo "  [Languages.stable]"
+              echo "    weight = -20000"
+              echo "    languageName = \"stable ($VERSION)\""
```

Comment:
@sophokles73 The backslash here is interpreted in terms of the Groovy string containing the bash script. In order to have proper escaping of the quote, two slashes are needed.
Another issue is the white space in the Groovy string before `#!/bin/bash` above, meaning bash wasn't actually used.

Both has been fixed in 825c1e1b8df1ed855d3d13f541b10d3afc93f08a.